src/Controls/src/Core/ShadowTypeConverter.cs

```diff
@@ -0,0 +1,141 @@
+﻿#nullable disable
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using Microsoft.Maui.Graphics;
+
+namespace Microsoft.Maui.Controls
+{
+	/// <summary>
+	/// Type converter for converting a properly formatted string to a Shadow.
+	/// </summary>
+	public class ShadowTypeConverter : TypeConverter
+	{
+		/// <summary>
+		/// Checks whether the given <paramref name="sourceType" /> is a string.
+		/// </summary>
+		/// <param name="context">The context to use for conversion.</param>
+		/// <param name="sourceType">The type to convert from.</param>
+		/// <returns></returns>
+		public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+			=> sourceType == typeof(string);
+
+		/// <summary>
+		/// Checks whether the given <paramref name="destinationType" /> is a Shadow.
+		/// </summary>
+		/// <param name="context">The context to use for conversion.</param>
+		/// <param name="destinationType">The type to convert to.</param>
+		/// <returns></returns>
+		public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+			=> destinationType == typeof(Shadow);
+
+		/// <summary>
+		/// Converts <paramref name="value" /> to a Shadow.
+		/// </summary>
+		/// <param name="context">The context to use for conversion.</param>
+		/// <param name="culture">The culture to use for conversion.</param>
+		/// <param name="value">The value to convert.</param>
+		/// <returns></returns>
+		/// <exception cref="ArgumentNullException">Thrown when <paramref name="value" /> is null.</exception>
+		/// <exception cref="InvalidOperationException">Thrown when <paramref name="value" /> is not a valid Shadow.</exception>
+		public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+		{
+			var strValue = value?.ToString();
+
+			if (strValue == null)
+			{
+				throw new ArgumentNullException(nameof(strValue));
+			}
+
+			var parts = strValue.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+
+			try
+			{
+				if (parts.Length == 3) // <color> | <float> | <float> e.g. #000000 4 4
+				{
+					var brush = new SolidColorBrush(Color.FromArgb(parts[0]));
+					var offsetX = float.Parse(parts[1], CultureInfo.InvariantCulture);
+					var offsetY = float.Parse(parts[2], CultureInfo.InvariantCulture);
+
+					return new Shadow
+					{
+						Brush = brush,
+						Offset = new Point(offsetX, offsetY)
+					};
+				}
+				else if (parts.Length == 4) // <float> | <float> | <float> | <color> e.g. 4 4 16 #000000
+				{
+					var offsetX = float.Parse(parts[0], CultureInfo.InvariantCulture);
+					var offsetY = float.Parse(parts[1], CultureInfo.InvariantCulture);
+					var radius = float.Parse(parts[2], CultureInfo.InvariantCulture);
+					var brush = new SolidColorBrush(Color.FromArgb(parts[3]));
```

Comment:
As in other converters, such as BrushTypeConverter, can we allow the use of hex, rgb etc colors?
https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/src/Core/Brush/BrushTypeConverter.cs#L125